nixos/modules/config/qt5.nix

```diff
@@ -0,0 +1,102 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.qt5;
+
+  isQGnome = cfg.platformTheme == "gnome" && cfg.style == "adwaita";
+  isQtStyle = cfg.platformTheme == "gtk2" && cfg.style != "adwaita";
+
+  packages = if isQGnome then [ pkgs.qgnomeplatform pkgs.adwaita-qt ]
+    else if isQtStyle then [ pkgs.qtstyleplugins ]
+    else throw "`qt5.platformTheme` ${cfg.platformTheme} and `qt5.style` ${cfg.style} are not compatable.";
+
+in
+
+{
+
+  options = {
+    qt5 = {
+
+      enable = mkEnableOption "Qt5 themeing configuration";
+
+      platformTheme = mkOption {
+        type = types.enum [
+          "gtk2"
+          "gnome"
+        ];
```

Comment:
What about other options, like qt5ct?

```
.../nixpkgs/nixpkgs $ nix-locate -r plugins/platformthemes.*.so | cut -d/ -f9 | sort -u
KDEPlasmaPlatformTheme.so
liblthemeengine.so
libqdeepin.so
libqgnomeplatform.so
libqgtk2.so
libqgtk3.so
libqt5ct.so
libqtlxqt.so
libqflatpak.so
libqxdgdesktopportal.so
```

(What exactly are [qxdgdesktopportal](https://github.com/qt/qtbase/tree/5.12/src/plugins/platformthemes/xdgdesktopportal) and qflatpak here? From a cursory look looks like it's an implementation detail for portals...)

Reply:
> (What exactly are qxdgdesktopportal and qflatpak here? From a cursory look looks like it's an implementation detail for portals...)

I'm guess that perhaps those get set environmentally within the flatpak sandbox, so nixos configuration would be needed there

> What about other options, like qt5ct?

It doesn't sound declarative 
```
Whether to enable the Qt5 Configuration Tool (qt5ct), a
program that allows users to configure Qt5 settings (theme,
font, icons, etc.) under desktop environments or window
manager without Qt integration.

And for the lxqt plugin, I've happened to find out today that their session script sets them manually.